test/autoload_test.rb

```diff
@@ -373,6 +377,27 @@
       bufname.must_equal File.join(@dir, "other.js")
     end
 
+    it "must edit ./other.js in non-readonly mode" do
+      touch File.join(@dir, "other.js")
+      Dir.mkdir File.join(@dir, "lib")
+      $vim.edit File.join(@dir, "lib", "CHANGELOG.txt")
+      $vim.command "Nedit ./other"
+      bufname = File.realpath($vim.echo(%(bufname("%"))))
+      bufname.must_equal File.join(@dir, "other.js")
+      $vim.echo(%(&readonly)).must_equal "0"
+      $vim.echo(%(&modifiable)).must_equal "1"
+    end
+
+    # it "must edit core modules in readonly mode" do
+    #   Dir.mkdir File.join(@dir, "lib")
+    #   $vim.edit File.join(@dir, "lib", "CHANGELOG.txt")
+    #   $vim.command "Nedit fs"
```

Comment:
I wrote out this test, and although it passes (when I uncomment it), it  seems to have side effects that cause 3 other tests to fail:

```
  1) Failure:
Autoloaded::Goto file#test_0017_must find when filetype is JavaScript and file new [/home/nate/programming/forks/vim-node/test/autoload_test.rb:186]:
--- expected
+++ actual
@@ -1 +1 @@
-"/tmp/d20160112-17778-1ru217k/node_modules/foo/index.js"
+"/tmp/d20160112-17778-1ru217k/index.js"


  2) Failure:
Autoloaded::Goto file with split#test_0001_must edit file in a new split [/home/nate/programming/forks/vim-node/test/autoload_test.rb:270]:
Expected: "1"
  Actual: "0"

  3) Failure:
Autoloaded::Goto file with tab#test_0001_must edit file in a new tab [/home/nate/programming/forks/vim-node/test/autoload_test.rb:286]:
Expected: "1"
  Actual: "0"
```

Any idea what the side effect could be, @moll? The other tests start passing if I uncomment the `Nedit fs` line.